lib/Runtime/Library/JsBuiltIn/JsBuiltIn.js

```diff
@@ -181,6 +181,229 @@
         }
     });
 
+    platform.registerChakraLibraryFunction("DefaultMergeSort", function(array, length) {
+        const compareArray = [];
+        for (let index = 0; index < length; ++index) {
+            __chakraLibrary.arrayCreateDataPropertyOrThrow(compareArray, index, "" + array[index]);
+        }
+
+        const buffer = [];
+        const stringBuffer = [];
+        let i = 1, j = 0, k = 0, bucketSize = 4, lastSize = 2, position = 0;
+
+        // handle pairs first
+        while (i < length) {
+            if (compareArray[i] < compareArray[i - 1]) {
+                const cache = array[i];
+                array[i] = array[i - 1];
+                array[i - 1] = cache;
+
+                const stringCache = compareArray[i];
+                compareArray[i] = compareArray[i - 1];
+                compareArray[i - 1] = stringCache;
+            }
+            i += 2;
+        }
+        const doubleLength = length * 2;
+
+        while (bucketSize < doubleLength) {
+            while (position < length) {
+                const left = position;
+                let right = position + bucketSize;
+                const mid = left + lastSize;
+
+                // perform a merge but only if it's necessary
+                if (mid < length && compareArray[mid] < compareArray[mid - 1]) {
+                    right = right < length ? right : length;
```

Comment:
nit: I think it would be a little cleaner to move `let right = position + bucketSize` down here so all the `right` computation is visually together and the `right` variable is scoped to the smallest block that uses it.

Reply:
done